nvflare/app_opt/script_executor.py

```diff
@@ -0,0 +1,181 @@
+                raise ValueError("Using FrameworkType.PYTORCH, but unable to import torch")
+        elif self._framework == FrameworkType.TENSORFLOW:
+            _, tf_ok = optional_import(module="tensorflow")
+            if tf_ok:
+                from nvflare.app_opt.tf.params_converter import (
+                    KerasModelToNumpyParamsConverter,
+                    NumpyToKerasModelParamsConverter,
+                )
+
+                self._from_nvflare_converter = NumpyToKerasModelParamsConverter(
+                    [AppConstants.TASK_TRAIN, AppConstants.TASK_VALIDATION]
+                )
+                self._to_nvflare_converter = KerasModelToNumpyParamsConverter(
+                    [AppConstants.TASK_TRAIN, AppConstants.TASK_SUBMIT_MODEL]
+                )
+                self._params_exchange_format = ExchangeFormat.NUMPY
+            else:
+                raise ValueError("Using FrameworkType.TENSORFLOW, but unable to import tensorflow")
+        elif self._framework == FrameworkType.NUMPY:
+            self._params_exchange_format = ExchangeFormat.NUMPY
+        elif self._framework == FrameworkType.RAW:
+            self._params_exchange_format = ExchangeFormat.RAW
+        else:
+            raise ValueError(f"Framework {self._framework} unsupported")
+
+    def add_to_fed_job(self, job, ctx, **kwargs):
+        """This method is used by Job API.
+
+        Args:
+            job: the Job object to add to
+            ctx: Job Context
+
+        Returns:
+
+        """
+        job.check_kwargs(args_to_check=kwargs, args_expected={"tasks": False})
+        tasks = kwargs.get("tasks", ["*"])
+
+        from_nvflare_id = None
+        to_nvflare_id = None
+
+        if self._from_nvflare_converter and self._to_nvflare_converter:
+            from_nvflare_id = job.add_component("from_nvflare", self._from_nvflare_converter, ctx)
+            to_nvflare_id = job.add_component("to_nvflare", self._to_nvflare_converter, ctx)
+
+        if self._launch_external_process:
+            from nvflare.app_common.executors.client_api_launcher_executor import ClientAPILauncherExecutor
+            from nvflare.app_common.launchers.subprocess_launcher import SubprocessLauncher
+            from nvflare.app_common.widgets.external_configurator import ExternalConfigurator
+            from nvflare.app_common.widgets.metric_relay import MetricRelay
+            from nvflare.fuel.utils.pipe.cell_pipe import CellPipe
+
+            component = CellPipe(
+                mode="PASSIVE",
+                site_name="{SITE_NAME}",
+                token="{JOB_ID}",
+                root_url="{ROOT_URL}",
+                secure_mode="{SECURE_MODE}",
+                workspace_dir="{WORKSPACE}",
+            )
+            pipe_id = job.add_component("pipe", component, ctx)
+
+            component = SubprocessLauncher(
+                script=self._command + " custom/" + os.path.basename(self._script) + " " + self._script_args,
+            )
```

Comment:
We need "add_resources" for this script?
Only if we do that, then this "custom" needs to be added.

it should be the full script name without any directories.
not sure if we should do `os.path.basename`

People can also put absolute path (/tmp/hello/cool) here, then in that case this "custom" is not fitted into it.
Also we could argue in the case of ScriptExecutor we only support relative path.

Reply:
`self._script` is added to resources in both cases at the end. `custom` doesn't need to be added, as adding the resources is copying to the custom folder. 

Full script name is supported, as when we copy to the custom folder we take the basename.

Reply:
got it, for script/file this behavior is clear. thanks!